src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -92,6 +93,44 @@ class TrajectoryPlannerTest : public testing::Test
 
         return acceleration;
     }
+
+    static std::vector<Vector> getDirectionVectorsFromPositionTrajectory(
+        PositionTrajectory_t* position_trajectory)
+    {
+        PositionTrajectoryElement_t* position_elements =
+            position_trajectory->trajectory_elements;
+        std::vector<Vector> unit_vectors;
+
+        for (unsigned int i = 0;
+             i < position_trajectory->path_parameters.num_segments - 1; i++)
+        {
+            const float delta_x =
+                position_elements[i + 1].position.x - position_elements[i].position.x;
+            const float delta_y =
+                position_elements[i + 1].position.y - position_elements[i].position.y;
+
+            // Copy data into the Vector
+            Vector direction = Vector(delta_x, delta_y).normalize(1);
+            unit_vectors.push_back(direction);
+        }
+
+        // Assume that the velocity of the final segment is in the direction of the
+        // previous
+        const float delta_x =
+            position_elements[position_trajectory->path_parameters.num_segments - 1]
+                .position.x -
+            position_elements[position_trajectory->path_parameters.num_segments - 2]
+                .position.x;
+        const float delta_y =
+            position_elements[position_trajectory->path_parameters.num_segments - 1]
+                .position.y -
+            position_elements[position_trajectory->path_parameters.num_segments - 2]
+                .position.y;
+
+        // The unit vector of the direction is 1/magnitide(vector) *vector
```

Comment:
nit: looks like a copied comment not needed here since normalization is abstracted away.

Reply:
Removed comment 👍